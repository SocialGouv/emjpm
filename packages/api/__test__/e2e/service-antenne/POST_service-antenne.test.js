//

const request = require("supertest");
const nodemailerMock = require("nodemailer-mock");
jest.setMock("nodemailer", nodemailerMock);

process.env.SMTP_FROM = "ne-pas-repondre@emjpm.gouv.fr";
process.env.APP_URL = "https://emjpm.gouv.fr";

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

beforeEach(async () => {
  nodemailerMock.mock.reset();
});

const defaultRegister = {
  contact_nom: "testMandaAd",
  contact_prenom: "testPrenom",
  etablissement: "",
  contact_email: "totoManda@toto.com",
  type: "service",
  adresse: "",
  code_postal: "75010",
  ville: "",
  telephone: "",
  tis: [1],
  service_id: 1
};

// eslint-disable-next-line no-unused-vars
const simpler = ({ created_at, ...props }) => props;

test("should create a mandataire with antenne", async () => {
  const token = await getTokenByUserType("service");
  const response = await request(server)
    .post("/api/v1/mandataires")
    .set("Authorization", "Bearer " + token)
    .send(defaultRegister);
  expect(response.body.success).toBe(true);
  expect(response.status).toBe(200);

  const lastInsert = await knex
    .table("mandataires")
    .orderBy("created_at", "desc")
    .first();
  expect(simpler(lastInsert)).toMatchSnapshot();
});
