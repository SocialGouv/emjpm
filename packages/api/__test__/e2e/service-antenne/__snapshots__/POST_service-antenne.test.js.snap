// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a mandataire with antenne 1`] = `
Object {
  "adresse": "",
  "antenne_id": 1,
  "code_postal": "75010",
  "contact_email": "totoManda@toto.com",
  "contact_nom": "testMandaAd",
  "contact_prenom": "testPrenom",
  "date_mesure_update": null,
  "dispo_max": null,
  "email_send": null,
  "etablissement": "",
  "genre": null,
  "id": 6,
  "mesures_en_attente": null,
  "mesures_en_cours": null,
  "nb_secretariat": null,
  "secretariat": null,
  "telephone": "",
  "telephone_portable": null,
  "user_id": 3,
  "ville": "",
  "zip": null,
}
`;
